### ISSUE TITLE
Targeting BMv2 psa_switch

### ISSUE THREAD

**Issue by CommitThis** [p4lang/p4c#2402]
Caveat: I haven't actually written a program to test this, however the docs don't seem to be clear on the matter.

The docs state that p4c can target BMv2 `simple_switch` but say nothing (that I can see) of the `psa_switch`. however there appear to be tests for the latter. Is targeting PSA possible, and if so could this be clarified in the documentation? Additionally, if it is possible, is there anything that needs to be specified e.g. a different "--arch".

I also built the Doxygen docs but I couldn't get the search to work. Incidentally, have you considered hosting the generated docs online somewhere?

Thanks

**Comment by jnfoster** [comment]
The p4c backend and support on Bmv2 for PSA are a work-in-progress. 

There is partial support, hence, the tests, but it is not complete. 

Nobody was actively working on this for a while, but it's been picked up in the past 2 months.

**Comment by CommitThis** [comment]
Thanks for clarifying. I agree it wouldn't be a good idea to put up docs prematurely - I think what I was getting at was if it was supported, add it in to the readme as that was the first place I looked (sorry I should have been more clear). I don't know whether or not it would be worthwhile mentioning that it _isn't_ supported there to avoid confusion, as was my case.

I've had a look at the link you provided and am relieved to see that the search works -- I did find it a bit laborious to make sure I had Doxygen installed, set the CMake flags and then build the documentation and not have search work (although I couldn't tell you why it didn't).
